### ISSUE TITLE
Traefik session stickiness based on Backend service cookie

### ISSUE THREAD

**Issue by semangard** [traefik/traefik#8641]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.containo.us) and didn't find any.

### What did you expect to see?

In order to load-balance properly request to our dockerized KeyCloak and improve its performances : [keycloak-documentation/sticky-sessions.adoc at master · keycloak/keycloak-documentation · GitHub 8](https://github.com/keycloak/keycloak-documentation/blob/main/server_installation/topics/clustering/sticky-sessions.adoc)

--> The load-balancing should be based on AUTH_SESSION_ID KeyCloak cookie whose format is 
`<session-id>.<owner-node-id>
`
Reason : request should be sent to the KeyCloak node identified by `<owner-node-id>`

**Comment by rtribotte** [comment]
Hello @semangard,

Thanks for your interest in Traefik!

The scenario described in the KeyCLoak documentation seems a bit unclear according to this conversation: https://github.com/keycloak/keycloak/discussions/8692.

The variant A scenario doesn't seem feasible without introducing something very specific to KeyCloak. It would mean both, to be able to parse the `AUTH_SESSION_ID` and to know which server corresponds to the `<owner-node-id>`.
And since, regarding the core features of Traefik (configuration aside), Traefik usually does not include anything that is not a widely used standard, variant A is not an acceptable scenario.

However, as a side note, it seems feasible to achieve this scenario via Header routing, although it would require having as many routers as KeyCloak servers.

The variant B scenario is achievable with sticky cookies.

In case I did not understand what you are expecting, please, feel free to reopen with more details.

**Comment by securite-apps** [comment]
Hello @rtribotte 

Indeed  Traefik should be able to parse the special `<session-id>.<owner-node-id`> syntax to read the ownerNodeId (this mapping between the cookie value and the node IP can also be kept into memory by Traefik), so it can route the subsequent requests to the optimal node.
Please note that we deployed KeyCloak and Traefik on Docker Swarm and this should be achieved dynamically.
To summarize, instead of using its own cookie, Traefik should be able to use keyCloak cookie.
